### PR TITLE
Call BroadcastNetworkFailure when the PlayerController gets deleted by the server.

### PR DESCRIPTION
**Pull request by Vatyx**
#### Description
If the server stops receiving heartbeats from the client, it close the client's connection and delete the PlayerController and all associated Actors. However the client could still be connected to SpatialOS when this happens. In this case, we should notify the client of a network failure and allow the developer to hand the fail condition.

#### Tests
Test locally and on Android

#### Primary reviewers
@improbable-valentyn @m-samiec

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp [hunk 1]
```diff
@@ -1012,6 +1012,12 @@ void USpatialReceiver::RemoveActor(Worker_EntityId EntityId)
 	{
 		// Force APlayerController::DestroyNetworkActorHandled to return false
 		PC->Player = nullptr;
+		
+		if (!NetDriver->IsServer())
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: Add a comment why this is needed and a TODO to remove this once UNR-3006 is done.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp [hunk 2]
```diff
@@ -1012,6 +1012,12 @@ void USpatialReceiver::RemoveActor(Worker_EntityId EntityId)
...
+		{
+			GEngine->BroadcastNetworkFailure(NetDriver->GetWorld(), NetDriver, ENetworkFailure::ConnectionLost, 
+							 TEXT("PlayerController %s deleted. Server believes we have been timed out."), *PC->GetName());
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: Couldn't this fire in the normal play dc flow - if the server deletes the pc before the client has dc-ed from spatial?

Comment (Vatyx) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: Is that an issue?

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: But wouldn't a real dc be caught by `USpatialReceiver::OnDisconnect` ? Which issues the same network failure broadcast.

**Pull request merged**